### PR TITLE
Feature/eng 486 update deployer with runtime changes

### PR DESCRIPTION
**Pull request by oddgrd**
## Description of change

This updates deployer to work with the changes we made to the runtime. So for legacy it starts the users project with the embedded runtime (and sends the path to the same runtime in the load request, but it isn't used in the legacy runtime load function), and for shuttle-next it installs and starts the shuttle-next runtime binary, and then sends the users compiled .wasm file in the load request.

TODO:

- [ ] Can probably clean up some of the code in deployer.
- [ ] Figure out if we fix the tests here or after merging/backporting changes from https://github.com/shuttle-hq/shuttle/pull/580

## How Has This Been Tested (if applicable)?

Most of the tests pass, but the ones that deal with a project crashing or stopping don't. This is likely solved in https://github.com/shuttle-hq/shuttle/pull/580, so I'm not sure if we need to do it in this PR.

### CODE REVIEW

File: common/src/storage_manager.rs [hunk 1]
```diff
@@ -2,14 +2,14 @@ use std::{fs, io, path::PathBuf};
...
+pub trait StorageManager: Sync + Send {
     /// Path for a specific service build files
-    fn service_build_path<S: AsRef<str>>(&self, service_name: S) -> Result<PathBuf, io::Error>;
+    fn service_build_path(&self, service_name: String) -> Result<PathBuf, io::Error>;
 
     /// Path to folder for storing deployment files
-    fn deployment_storage_path<S: AsRef<str>>(
+    fn deployment_storage_path(
         &self,
-        service_name: S,
```
Comment (chesedo) [inline common/src/storage_manager.rs:12]: I assume this gave some issues? Maybe we rather want to use `&str`?

File: runtime/src/legacy/mod.rs [hunk 2]
```diff
@@ -47,21 +47,28 @@ use self::args::Args;
...
-) {
+pub async fn start(loader: impl Loader<ProvisionerFactory> + Send + 'static) {
     let args = Args::parse();
     let addr = SocketAddr::new(Ipv4Addr::LOCALHOST.into(), args.port);
 
     let provisioner_address = args.provisioner_address;
     let mut server_builder =
         Server::builder().http2_keepalive_interval(Some(Duration::from_secs(60)));
 
+    let storage_manager: Arc<dyn StorageManager> = match args.storage_manager_type {
```
Comment (chesedo) [inline runtime/src/legacy/mod.rs:58]: Wow, nice!!

**Review by chesedo — COMMENTED** [review]
Nice solve O!

File: runtime/src/legacy/mod.rs (hunk 2 above)
Comment (oddgrd) [reply runtime/src/legacy/mod.rs:58]: Thanks!

File: common/src/storage_manager.rs (hunk 1 above)
Comment (oddgrd) [reply common/src/storage_manager.rs:12]: Yes, I believe having generics in the methods is not object safe, but we can probably use &str, let me check!

Comment (oddgrd) [reply common/src/storage_manager.rs:12]: Yeah, generic types in methods are not object safe: https://doc.rust-lang.org/reference/items/traits.html#object-safety

**Review by chesedo — APPROVED** [review]
Thanks for all the effort on this one @oddgrd!!

LGTM

**Pull request merged**